### ISSUE TITLE
Reproduction Issue of genPPL Regarding auto-BLEU-2

### ISSUE THREAD

**Issue by rerorero6034** [slp-rl/slamkit#19]
Hello author,
I am encountering an issue with the ​​auto-BLEU-2​​ values while trying to reproduce the results using your repository. In your paper, you mentioned that to ensure a fair comparison of ​​genPPL​​, the auto-BLEU-2 values for all models were kept similar. Could you please explain how you constrained the auto-BLEU-2 values? Was it done by filtering out data points where the value exceeded a certain threshold?
When I trained the model using only the ​​Libri-light​​ dataset and filtered out data with auto-BLEU-2 > 0.1, the genPPL score was 223, while the auto-BLEU-2 score was 0.015.
Additionally, I would like to ask about handling the long audio samples in the Libri-light dataset. Directly extracting features and training on the unprocessed long sequences consumes significant GPU memory. How did you process these long data samples? (In my own training, I discarded the longest samples.)

I would greatly appreciate it if you could provide some guidance.

**Comment by rerorero6034** [comment]
​​Additional Information:​​
When using the official model open-sourced from https://huggingface.co/slprl/slam, the evaluation results still do not match those reported in the paper.
​​Specific commands and evaluation results:​​
```
python cli/eval.py \
    tokeniser=unit_hubert_25 \
    metric=asr_perplexity \
    model.pretrained_model=/slamkit/outputs/slam_official \
    metric.data_path="/slamkit/dataset/sSC/*_correct.wav" \
    vocoder=vocoder_hubert_25 \
    metric.used_token_modality=speech
```
​​asr_perplexity​​: 54.74813461303711 (Reported in paper: 62.8)
​​auto-bleu-2​​: 0.06584221124649048 (Reported in paper: 0.0338)

-----------------------------------------------------------------------
```
python cli/eval.py \
    tokeniser=unit_hubert_25 \
    metric=sblimp \
    model.pretrained_model=/slamkit/outputs/slam_official \
    metric.data_path=/slamkit/dataset/sblimp_prepared_for_eval \
    metric.used_token_modality=speech \
    metric.subfolder=False
```
​​sBLIMP​​: 0.49675002694129944 (Reported in paper: 0.5586)

**Comment by avishaiElmakies** [comment]
Hi! Thanks for the intrest in our work!

Regarding your results for asr ppl using slam. If i remember correctly we also use repetition penalty of 1.1 when generating. This should give you results similar to the one in the paper.

Regarding sblimp results with slam. This is kinda hard to debug without you giving us more information about how sblimp is setup for you. More info can be found here https://github.com/slp-rl/slamkit/issues/8

regarding your first msg. We didn't constrainted the BLEU score with code or something. What we meant was that we wanted the BLEU score for our models to be similar before comparing the asr ppl score. It is more of grounding metric to make sure that that the asr ppl score makes sense. For example let's look at model A with BLEU 0.2 and asr ppl of 10,model B has BLEU 0.01 with asr ppl of 60. Is model A really better with generation? 
When both models have similar BLEU it makes much more sense to compare asr ppl. So we didn't constraint it with data or anything. At most we used different generations params to make bleu more comparable. Especially for models after dpo where we needed to add repetition penalty to make the BLEUs more comparable. Other than that we ddin't do anything except what is written in the paper (if i remember correctly we did filter out samples with high BLEU for the dpo step, but not the pretraining step) 

Regarding the long audio samples in librilight. We used the processing script and metadata given for the dataset to create 80 seconds long files(using the vad metadata), And used it to train our model. There will be some files which are still very long, i believe we ignored or turnicated them.
